api/tests/unit/integrations/lead_tracking/hubspot/_hubspot_responses.py

```diff
@@ -0,0 +1,106 @@
+from datetime import datetime
+
+from dateutil.tz import tzlocal
+
+
+class DummyHubspotResponse:
+    """
+    Dummy class to replicate the to_dict() method of the Hubspot response classes.
+    """
+
+    def __init__(self, data: dict) -> None:
+        self.data = data
+
+    def to_dict(self) -> dict:
+        return self.data
+
+
+def generate_get_company_by_domain_response(
+    name: str, domain: str
+) -> DummyHubspotResponse:
+    """
+    Generate a sample response given by the Hubspot API when searching for a company by domain.
+
+    This response was retrieved from the API directly, and then modified to allow us to set the
+    certain properties dynamically.
+    """
+
+    return DummyHubspotResponse(
+        data={
+            "paging": None,
+            "results": [
+                {
+                    "archived": False,
+                    "archived_at": None,
+                    "created_at": datetime(
+                        2024, 1, 25, 21, 48, 28, 655000, tzinfo=tzlocal()
+                    ),
+                    "id": "9765318341",
+                    "properties": {
+                        "createdate": "2024-01-25T21:48:28.655Z",
```

Comment:
I know its not your fault, but it does seem weird that `created_at` is a `datetime` while `createdate` is a `str`.